submodules/LIB/scripts/setup_caravel.py

```diff
@@ -0,0 +1,54 @@
+import shutil
+import os
+import re
+import sys
+
+
+def copy_directory(src_path, dest_path, new_directory_name):
+    # Check if the destination directory exists, if not create it
+    if not os.path.exists(dest_path):
+        os.makedirs(dest_path)
+
+    # Create the full path for the renamed directory in the destination
+    new_dest_path = os.path.join(dest_path, new_directory_name)
+
+    # Copy the entire directory from src_path to new_dest_path
+    shutil.copytree(src_path, new_dest_path)
+
+
+if __name__ == "__main__":
+    if len(sys.argv) != 4:
+        print(
+            "Usage: python script_name.py <source_directory> <destination_directory> <new_directory_name>"
+        )
+        sys.exit(1)
+
+
+source_directory = sys.argv[1]
+destination_directory = sys.argv[2]
+new_directory_name = sys.argv[3]
+copy_directory(source_directory, destination_directory, new_directory_name)
+
+
```

Comment:
this part detects the verilog files, and detects what is the top model. The top model is the one with the VDD and GND pins i assume.